azurerm/internal/services/mssql/parse/mssql.go

```diff
@@ -29,6 +29,18 @@ type MsSqlDatabaseExtendedAuditingPolicyId struct {
 	ResourceGroup string
 }
 
+func NewMsSqlDatabaseID(resourceGroup, msSqlServer, name string) MsSqlDatabaseId {
+	return MsSqlDatabaseId{
+		ResourceGroup: resourceGroup,
+		MsSqlServer:   msSqlServer,
+		Name:          name,
+	}
```

Comment:
Same here, shouldn't we expose a new `struct` with a more descriptive name?
```suggestion
	return SynapseSqlPollDatabaseId{
		ResourceGroup: resourceGroup,
		SynapseSqlPoolServer:   synapseSqlPoolServer,
		Name:          name,
	}
```